xilem/src/view/split.rs

```diff
@@ -0,0 +1,280 @@
+    /// Vertical split axis means that the children are up and down.
+    ///
+    /// The default split point is horizontal.
+    pub fn split_axis(mut self, axis: Axis) -> Self {
+        self.split_axis = axis;
+        self
+    }
+
+    /// Set the split point as a fraction of the split axis.
+    ///
+    /// The value must be between `0.0` and `1.0`, inclusive.
+    /// The default split point is `0.5`.
+    pub fn split_point(mut self, split_point: f64) -> Self {
+        assert!(
+            (0.0..=1.0).contains(&split_point),
+            "split_point must be in the range [0.0-1.0]!"
+        );
+        self.split_point = split_point;
+        self
+    }
+
+    /// Set the minimum size for both sides of the split axis.
+    ///
+    /// The value must be greater than or equal to `0.0`.
+    /// The value will be rounded up to the nearest integer.
+    pub fn min_size(mut self, first: f64, second: f64) -> Self {
+        assert!(first >= 0.0);
+        assert!(second >= 0.0);
+        self.min_size = (first.ceil(), second.ceil());
+        self
+    }
+
+    /// Set the size of the splitter bar.
+    ///
+    /// The value must be positive or zero.
+    /// The value will be rounded up to the nearest integer.
+    /// The default splitter bar size is `6.0`.
+    pub fn bar_size(mut self, bar_size: f64) -> Self {
+        assert!(bar_size >= 0.0, "bar_size must be 0.0 or greater!");
+        self.bar_size = bar_size.ceil();
+        self
+    }
+
+    /// Set the minimum size of the splitter bar area.
+    ///
+    /// The minimum splitter bar area defines the minimum size of the area
+    /// where mouse hit detection is done for the splitter bar.
+    /// The final area is either this or the splitter bar size, whichever is greater.
+    ///
+    /// This can be useful when you want to use a very narrow visual splitter bar,
+    /// but don't want to sacrifice user experience by making it hard to click on.
+    ///
+    /// The value must be positive or zero.
+    /// The value will be rounded up to the nearest integer.
+    /// The default minimum splitter bar area is `6.0`.
+    pub fn min_bar_area(mut self, min_bar_area: f64) -> Self {
+        assert!(min_bar_area >= 0.0, "min_bar_area must be 0.0 or greater!");
+        self.min_bar_area = min_bar_area.ceil();
+        self
+    }
+
+    /// Set whether the split point can be changed by dragging.
+    ///
+    /// The default is false.
+    pub fn draggable(mut self, draggable: bool) -> Self {
+        self.draggable = draggable;
+        self
+    }
+
+    /// Set whether the splitter bar is drawn as a solid rectangle.
+    ///
+    /// If this is `false` (the default), the bar will be drawn as two parallel lines.
+    pub fn solid_bar(mut self, solid: bool) -> Self {
+        self.solid_bar = solid;
+        self
+    }
+}
+
+const CHILD1_VIEW_ID: ViewId = ViewId::new(0);
+const CHILD2_VIEW_ID: ViewId = ViewId::new(0);
```

Comment:
> *But* they need to be different, otherwise all messages will be routed to the first child1.

I think that went probably under the radar.
So to be more explicit:
`CHILD1_VIEW_ID` needs to be something different than `CHILD2_VIEW_ID`, so e.g.

```suggestion
const CHILD1_VIEW_ID: ViewId = ViewId::new(5436781);
const CHILD2_VIEW_ID: ViewId = ViewId::new(5436782);
```

(btw. just my preference, and more of a nit, you could use `match` in `View::message`)

Reply:
Right, that was an oversight, I've pushed an update. Is there any particular reason to using pseudo-random numbers rather than simply 0 and 1?

Reply:
The reason you'd want to use a random key would be to be able to identify after-the-fact which `View`s were used in the tree just by inspecting the view id path. However, we've not done that for any other Views.

Reply:
Alright, should I change it to be 0 and 1 then, or is it fine as is?

Reply:
Yeah, 0 and 1 is probably best.